src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 32;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private int size;
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        this.size = 0;
+        this.table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeeded();
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
```

Comment:
This is not **hash** - this is **index**
And extract code to getIndex method (now you have code repeating)